doc/libpmemobj.3.md

```diff
@@ -1644,26 +1639,99 @@ Using **TX_PARAM_MUTEX** or **TX_PARAM_RWLOCK** means that at the beginning of a
+
+**pmemobj_tx_begin**() can also be called within an open transaction on the same memory pool if the current stage of the open transaction is in the **TX_STAGE_NONE** or **TX_STAGE_WORK**. In that case the call does not start any new transaction.
+Instead, the statements of the inner transaction become statements of proper stages of the outer transaction, committing the inner transaction does not commit the outer transaction, and errors in the inner transaction are propagated up to the outermost level resulting in the interruption of the entire transaction.
+If the call succeed, the transaction stage changes to **TX_STAGE_WORK** and the function returns zero. Otherwise, thes stage changes to **TX_STAGE_ONABORT** and an error number is returned.
+Calling **pmemobj_tx_begin**() when the open transaction is in a stage other than **TX_STAGE_NONE** and **TX_STAGE_WORK** fails and aborts the whole transaction.
+
+There are two significant differences between adding code to the **TX_ONCOMMIT**/**TX_ONABORT**/etc. sections of the inner transaction and adding stage callbacks to the inner transaction:
+
++  Registered callback functions are executed only in the most outer transaction (even if registered in the inner one).
 
 +  There can be only one callback in the whole transaction (it can't be changed in the inner transaction).
 
-Note that **TX_PARAM_CB** does not replace **TX_ONCOMMIT**/**TX_ONABORT**/etc. macros. They can be used together - a callback will be executed *before*
-**TX_ONCOMMIT**/**TX_ONABORT**/etc. section.
+**TX_PARAM_CB** can be used when the code dealing with transaction stage changes is shared between multiple users or when it must be executed only in the outer transaction. For example it can be very useful when application must synchronize persistent and transient state.
+
+To use a transaction, a *transaction context* must be present in the current thread. If none exists, a default one is automatically created.
+The transaction context is reused within a thread until either a) the application closes or the thread finishes, or b) it is explicitly changed using **pmemobj_tx_ctx_set**().
+
+```c
+int pmemobj_tx_ctx_new();
+int pmemobj_tx_ctx_delete(struct pobj_tx_ctx *ctx);
+int pmemobj_tx_ctx_set(struct pobj_tx_ctx *new_ctx, struct pobj_tx_ctx **old_ctx);
+```
+
+**pmemobj_tx_ctx_new**() creates a new transaction context.
 
-**TX_PARAM_CB** can be used when the code dealing with transaction stage changes is shared between multiple users or when it must be executed only in the outer
-transaction. For example it can be very useful when application must synchronize persistent and transient state.
+**pmemobj_tx_ctx_delete**() frees the given transaction context.
+
+**pmemobj_tx_ctx_set**() sets the current transaction context to *new_ctx*, storing the previous context under the *old_ctx* pointer unless the latter is NULL.
```

Comment:
" unless the latter is NULL." is a pleonasm. You can't store something if the said something doesn't exist.